src/imageops/colorops.rs

```diff
@@ -556,7 +545,38 @@ where
 mod test {
 
     use super::*;
-    use crate::ImageBuffer;
+    use crate::{GrayImage, ImageBuffer};
+
+    macro_rules! assert_pixels_eq {
```

Comment:
Why does this need to be a macro?

Reply:
I really didn't mind it following `assert_eq!` more closely in form. That said it does evaluate its arguments twice, now that you mention it.

Reply:
I have copy-paste `pixels_diffs` and the macro from the `affine.rs` file because `assert_pixel_eq!` is handy, but if it's a problem I can rewrite it as a function or use `assert_eq!`.

Reply:
I gravitate towards 'separate PR' for that change. The specific portion `diff(a, b) -> impl Iterator` could even be useful as a public function.

Reply:
I can help to work on that if you want.